OpenRA.Mods.Common/Widgets/Logic/ServerListLogic.cs

```diff
@@ -604,13 +604,14 @@ void SelectServer(GameServer server)
 				foreach (var option in kv.Value)
 				{
 					var o = option;
+					var playerName = o.IsBot ? FluentProvider.GetMessage(o.Name) : o.Name;
```

Comment:
I don't think this is going to work for maps that define their own custom bots.

It will need to query the message from the MapPreview instead, which then needs appropriate RC support to be implemented.

Reply:
good catch, it looks like MapPreviewWidget is also wrong